doc/api/tls.md

```diff
@@ -918,81 +890,87 @@ added: v0.11.13
-    several well known "root" CAs (like VeriSign) will be used. These are used
-    to authorize connections.
-  * `crl` {string|string[]} Either a string or array of strings of PEM encoded
-    CRLs (Certificate Revocation List).
-  * `ciphers` {string} A string describing the ciphers to use or exclude.
-    Consult
-    <https://www.openssl.org/docs/man1.0.2/apps/ciphers.html#CIPHER-LIST-FORMAT>
-    for details on the format.
-  * `honorCipherOrder` {boolean} If `true`, when a cipher is being selected,
-    the server's preferences will be used instead of the client preferences.
+  * `pfx` {string|Buffer} Optional PFX or PKCS12 encoded private key and
+    certificate chain. `pfx` is an alternative to providing `key` and `cert`
+    individually. PFX is usually encrypted, if it is, `passphrase` will be used
+    to decrypt it.
+  * `key` {string|string[]|Buffer|Buffer[]|Object[]} Optional private keys in
+    PEM format. Single keys will be decrypted with `passphrase` if necessary.
+    Multiple keys, probably using different algorithms, can be provided either
+    as an array of unencrypted key strings or buffers, or an array of objects in
+    the form `{pem: <string|buffer>, passphrase: <string>}`. The object form can
+    only occur in an array, and it _must_ include a passphrase, even if key is
+    not encrypted.
+  * `passphrase` {string} Optional shared passphrase used for a single private
+    key and/or a PFX.
+  * `cert` {string|string[]|Buffer|Buffer[]} Optional cert chains in PEM format.
+    One cert chain should be provided per private key. Each cert chain should
+    consist of the PEM formatted certificate for a provided private `key`,
+    followed by the PEM formatted intermediate certificates (if any), in order,
+    and not including the root CA (the root CA must be pre-known to the peer,
+    see `ca`).  When providing multiple cert chains, they do not have to be in
+    the same order as their private keys in `key`. If the intermediate
+    certificates are not provided, the peer will not be able to validate the
+    certificate, and the handshake will fail.
+  * `ca`{string|string[]|Buffer|Buffer[]} Optional CA certificates to trust.
+    Default is the well-known CAs from Mozilla. When connecting to peers that
+    use certificates issued privately, or self-signed, the private root CA or
+    self-signed certificate must be provided to verify the peer.
+  * `crl` {string|string[]|Buffer|Buffer[]} Optional PEM formatted
+    CRLs (Certificate Revocation Lists).
+  * `ciphers` {string} Optional cipher suite specification, replacing the
+    default.  For more information, see [modifying the default cipher suite][].
+  * `honorCipherOrder` {boolean} Attempt to use the server's cipher suite
+    preferences instead of the client's. When `true`, causes
+    `SSL_OP_CIPHER_SERVER_PREFERENCE` to be set in `secureOptions`, see
+    [OpenSSL Options][] for more information.
```

Comment:
I agree its odd. In the process of documenting the existing behaviour, I found many odd and irregular aspects of the TLS API, but this PR documents TLS "as is". I definitely encourage you to PR a fix to move the setting of the default out of `function Server()` and into `_tls_common` where it can apply to all TLS sockets, and be based on the `.isServer` property. If you don't, I will. I consider completing the docs to be step 1 to cleaning up the API. In the absence of docs, its hard for reviewers to understand why behaviour is inconsistent and needs to be fixed.